internal/overlord/servstate/handlers.go

```diff
@@ -601,6 +612,31 @@ func (s *serviceData) stop() error {
 	return nil
 }
 
+func (s *serviceData) waitProcessGroup() error {
+	var (
+		status syscall.WaitStatus
+		rusage syscall.Rusage
+		pid1   int
+		err    error
+	)
+	for {
+		// __WALL (since Linux 2.4)
+		// Wait for all children, regardless of type ("clone" or "non-clone").
+		pid1, err = syscall.Wait4(-s.cmd.Process.Pid, &status, syscall.WALL, &rusage)
```

Comment:
A couple of things:

* The docs say the wait functions check if `status` and `rusage` are not NULL, so we can just pass these in as `nil`, seeing we're not using the results.
* Also, let's rename `pid1` to `wpid` (what the `Wait4` docs call it).
* It's a bit simpler to use `wpid, err := syscall.Wait4(...)` and check err inside the loop to avoid having to define the vars at the top.
* Do we need to handle `EINTR` still?
* I think these scattered comments could be tidied up a bit. The `__WALL` comment and the die.net link probably aren't needed, for example.

Reply:
Agreed on the points and adjusted the code, please have a look. For EINTR, it is about pebble receiving a signal that is not blocked and I don't think we need it in this case. From the documentation:
`If a blocked call to one of the following interfaces is interrupted by a signal handler, then the call will be automatically restarted after the signal handler returns if the SA_RESTART flag was used; otherwise the call will fail with the error EINTR.`